ui-app/client/src/components/PhoneNumber/PhoneNumber.tsx

```diff
@@ -0,0 +1,653 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+	addListener,
+	addListenerAndCallImmediately,
+	getDataFromPath,
+	getPathFromLocation,
+	PageStoreExtractor,
+	setData,
+} from '../../context/StoreContext';
+import { ComponentPropertyDefinition, ComponentProps } from '../../types/common';
+import { Component } from '../../types/common';
+import { propertiesDefinition, stylePropertiesDefinition } from './phoneNumberProperties';
+import PhoneNumberStyle from './PhoneNumberStyle';
+import useDefinition from '../util/useDefinition';
+import { duplicate, isNullValue } from '@fincity/kirun-js';
+import { processComponentStylePseudoClasses } from '../../util/styleProcessor';
+import { STORE_PATH_FUNCTION_EXECUTION } from '../../constants';
+import { flattenUUID } from '../util/uuid';
+import { runEvent } from '../util/runEvent';
+import { validate } from '../../util/validationProcessor';
+import CommonInputText from '../../commonComponents/CommonInputText';
+import { styleDefaults } from './phoneNumberStyleProperties';
+import { IconHelper } from '../util/IconHelper';
+import { Dropdown, DropdownOption, DropdownOptions } from './components/Dropdown';
+import { COUNTRY_LIST } from './components/listOfCountries';
+import { SubHelperComponent } from '../HelperComponents/SubHelperComponent';
+
+interface mapType {
+	[key: string]: any;
+}
+
+function PhoneNumber(props: ComponentProps) {
+	const [focus, setFocus] = React.useState(false);
+	const [validationMessages, setValidationMessages] = React.useState<Array<string>>([]);
+	const mapValue: mapType = {
+		UNDEFINED: undefined,
+		NULL: null,
+		ENMPTYSTRING: '',
+		ZERO: 0,
+	};
+	const {
+		definition: { bindingPath },
+		definition,
+		pageDefinition: { translations },
+		locationHistory,
+		context,
+	} = props;
+	const pageExtractor = PageStoreExtractor.getForContext(context.pageName);
+	const {
+		properties: {
+			updateStoreImmediately: upStoreImm,
+			removeKeyWhenEmpty,
+			emptyValue,
+			supportingText,
+			readOnly,
+			defaultValue,
+			label,
+			noFloat,
+			onEnter,
+			validation,
+			placeholder,
+			messageDisplay,
+			autoComplete,
+			onClear,
+			onChange,
+			autoFocus,
+			designType,
+			colorScheme,
+			hideClearButton,
+			maxChars,
+			onFocus,
+			onBlur,
+			countries,
+			topCountries,
+			orderBy,
+			format,
+			storeFormatted,
+			seperator,
+			isSearchable,
+			searchLabel,
+			clearSearchTextOnClose,
+			noCodeForFirstCountry,
+		} = {},
+		stylePropertiesWithPseudoStates,
+		key,
+	} = useDefinition(
+		definition,
+		propertiesDefinition,
+		stylePropertiesDefinition,
+		locationHistory,
+		pageExtractor,
+	);
+	const [value, setValue] = React.useState<string>(defaultValue ?? '');
```

Comment:
code is failing when i give number instead of string in default value 
should default value be string